panel/widgets/select.py

```diff
@@ -654,6 +673,73 @@ def _update_options_programmatically(self):
                 self.value = original_values
                 raise
 
+    @classmethod
+    def create_options_from_dataframe(cls, data: pd.DataFrame, columns: List|None=None)->Dict|List:
```

Comment:
I do not think this should be public.

Reply:
The benefit of making it public is that users have more options for controlling performance. With the `create_from_dataframe` option there is no caching and the call will potentially be quite costly if the dataframe is large.

I would argue it should be public.